src/braket/circuits/gates.py

```diff
@@ -3847,6 +3847,77 @@ def pulse_gate(
 Gate.register_gate(PulseGate)
 
 
+class Barrier(Gate):
+    r"""Barrier gate."""
+
+    def __init__(self, qubit_count):
+        super().__init__(qubit_count=qubit_count, ascii_symbols=["||" for i in range(qubit_count)])
+
+    def bind_values(self, **kwargs) -> Any:
+        raise NotImplementedError
+
+    @property
+    def _qasm_name(self) -> str:
+        return "barrier"
+
+    def __hash__(self):
+        return hash((self.name, self.qubit_count))
+
+    @staticmethod
+    @circuit.subroutine(register=True)
+    def barrier(targets: QubitSetInput) -> Instruction:
+        r"""Barrier gate.
+
+        Args:
+            targets (QubitSetInput): Target qubit(s)
+
+        Examples:
+            >>> circ = Circuit().barrier(targets=[0, 1, 2])
+        """
+        return Instruction(Barrier(len(targets)), target=targets)
+
+
+Gate.register_gate(Barrier)
+
+
+class Delay(Gate):
+    r"""Delay gate. Applies delay in seconds."""
+
+    def __init__(self, qubit_count, duration):
+        super().__init__(
+            qubit_count=qubit_count,
+            ascii_symbols=[f"delay({duration})" for i in range(qubit_count)],
+        )
+        self.duration = duration
+
+    def bind_values(self, **kwargs) -> Any:
+        raise NotImplementedError
+
+    @property
+    def _qasm_name(self) -> str:
+        return f"delay[{self.duration} s]"
+
+    def __hash__(self):
+        return hash((self.name, self.qubit_count, self.duration))
+
+    @staticmethod
+    @circuit.subroutine(register=True)
+    def delay(targets: QubitSetInput, duration: float) -> Instruction:
+        r"""Delay gate. Applies delay in seconds.
+
+        Args:
+            targets (QubitSetInput): Target qubit(s)
+            duration (float): Delay(in seconds).
+
+        Examples:
+            >>> circ = Circuit().delay(targets=[0, 1, 2], duration = 30)
```

Comment:
Just a shorter delay, to make the example more realistic:
```suggestion
            >>> circ = Circuit().delay(targets=[0, 1, 2], duration=30e-9)
```